components/raftstore/src/store/worker/life.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::{Engines, KvEngine, RaftEngine};
+
+/// An assistant to help manage the lifecycle of engines.
+///
+/// Currently, it only enables manual compaction jobs before starting the
+/// engines and disables manual compaction jobs before shutting down the
+/// engines.
+pub struct Assistant<E, R>
+where
+    E: KvEngine,
+    R: RaftEngine,
+{
+    _name: String,
+    engines: Engines<E, R>,
+}
+
+impl<E, R> Assistant<E, R>
+where
+    E: KvEngine,
+    R: RaftEngine,
+{
+    pub fn new<S: Into<String>>(name: S, engines: Engines<E, R>) -> Self {
+        Self {
+            _name: name.into(),
+            engines,
+        }
+    }
+
+    pub fn start(&mut self) {
```

Comment:
the start/stop naming is kind of misleading. 
Here there's no background task running after this start() is called.